x/shared/keeper/msg_server_update_param_test.go

```diff
@@ -180,3 +214,49 @@ func TestMsgUpdateParam_UpdateGracePeriodEndOffsetBlocks(t *testing.T) {
 	// Ensure the other parameters are unchanged
 	testkeeper.AssertDefaultParamsEqualExceptFields(t, &defaultParams, res.Params, "GracePeriodEndOffsetBlocks")
 }
+
```

Comment:
Doesn't have to be done in this PR but can you please create a new file with a TODO_TEST and explanation of using `NewCompleteIntegrationApp` to testing the entire supplier lifecycle?